compile/compile.go

```diff
@@ -210,17 +219,83 @@ func (c *Compiler) WithMetadata(metadata *map[string]interface{}) *Compiler {
 	return c
 }
 
+func addEntrypointsFromAnnotations(c *Compiler, ar []*ast.AnnotationsRef) error {
+	for _, ref := range ar {
+		var entrypoint ast.Ref
+		scope := ref.Annotations.Scope
+
+		if ref.Annotations.Entrypoint {
+			// Build up the entrypoint path from either package path or rule.
+			switch scope {
+			case "package":
+				if p := ref.GetPackage(); p != nil {
+					entrypoint = p.Path
+				}
+			case "rule":
+				if r := ref.GetRule(); r != nil {
+					entrypoint = r.Ref()
+				}
+			default:
+				continue // Wrong scope type. Bail out early.
+			}
+
+			// Get a slash-based path, as with a CLI-provided entrypoint.
+			slashPath, err := entrypoint.Ptr()
```

Comment:
Things I'd test here are edge cases involving numbers in refs. Currently, only this should be possible:

```rego
package foo
#METADATA
entrypoint: true
bar.baz[1.0] = true
```
does it become the correct entrypoint? I'm wondering because we've had `array/0/foo` issues with the storage API before, and this is _somewhat_ related... but it's an edge case for sure.

Reply:
It errors currently, which is probably _not_ what we want. :sweat_smile:

Reply:
Okay, after borrowing the path escaping logic from `storage.path`, the above `bar.baz[1.0]` example would generate an entrypoint of `foo/bar/baz/1.0`

For longer paths that want a fully-variable part in the middle, they'll have to convert the variable to a string first. From what I can tell, the logic I borrowed from `storage` is fine, so long as only the last path segment in the ref is variable.